### DISCUSSION TITLE
OWL for self guided robot

### DISCUSSION THREAD

**Discussion by bartdereu** [geezacoleman/OpenWeedLocator#82] [General]
Hi all!

I'm in the process of building a weeding robot for my corn crops. Next to beeing a farmer i also run a software company , and i have been building self guided robots over a decade ago. The knowledge about that is not a real issue. I want to build a chemical free robot
that mechanically removes weeds . 

What's new for me, is the weed detection, so i was excited to see that OWL existed . My simple question is : can i use OWL to get coordinates of the weeds and send them to the computer that handles the mechanical arms? would be awesome.

Thanks in advance!

**Comment by geezacoleman** [comment]
That sounds very interesting - thanks for sharing the idea for your robot and using the OWL. Definitely keep us updated as you progress and feel free to ask if you run into any issues.

It depends what you mean by coordinates - if it's pixel coordinates of each weed (i.e. where in each image is the centre of the detection) then that is quite straightforward, just checkout [L279-293 in owl.py](https://github.com/geezacoleman/OpenWeedLocator/blob/main/owl.py#L279-L293) where the algorithm returns `weedCentres`. 

e.g.:
```
if algorithm == 'gog':
        cnts, boxes, weedCentres, imageOut = weed_detector.inference(frame.copy(), confidence=confidence, filter_id=63)
```

You can use this list to aim the robotic arm relative to each frame. For this type of targeting it would be important to [calibrate the ](https://docs.opencv.org/4.x/dc/dbb/tutorial_py_calibration.html) image so that the distortion on the edges doesn't impact the accuracy of the arm. 

GPS coordinates are slightly more difficult, but there is work being done on that at the moment.
